src/main/java/de/tum/in/www1/artemis/web/rest/TeamResource.java

```diff
@@ -326,4 +331,53 @@ public TeamResource(TeamRepository teamRepository, TeamService teamService, Cour
         return ResponseEntity.ok().body(destinationTeams);
     }
 
+    /**
+     * GET /courses/:courseId/teams/:teamShortName/with-exercises-and-participations : get course "id" with all released exercises in which the team "teamShortName" exists
+     * together with its participations for those exercises
+     *
+     * @param courseId id of the course
+     * @param teamShortName short name of the team (all teams with the short name in the course are seen as the same team)
+     * @return Course with exercises and participations for the team
+     */
+    @GetMapping(value = "/courses/{courseId}/teams/{teamShortName}/with-exercises-and-participations")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    public ResponseEntity<Course> getCourseWithExercisesAndParticipationsForTeam(@PathVariable Long courseId, @PathVariable String teamShortName) {
+        log.debug("REST request to get Course {} with exercises and participations for Team with short name {}", courseId, teamShortName);
+        Course course = courseService.findOneWithExercises(courseId);
```

Comment:
Please use a new method called `courseService.findOneWithTeamExercises` which filters only for team exercises

Reply:
Done